components/dashboard/src/menu/Menu.tsx

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { User } from "@gitpod/gitpod-protocol";
+import { useContext, useEffect, useMemo, useState } from "react";
+import { Link } from "react-router-dom";
+import { useLocation } from "react-router";
+import { Location } from "history";
+import { countries } from "countries-list";
+import gitpodIcon from "../icons/gitpod.svg";
+import { getGitpodService, gitpodHostUrl } from "../service/service";
+import { useCurrentUser } from "../user-context";
+import { useBillingModeForCurrentTeam, useCurrentTeam, useTeamMemberInfos } from "../teams/teams-context";
+import ContextMenu from "../components/ContextMenu";
+import Separator from "../components/Separator";
+import PillMenuItem from "../components/PillMenuItem";
+import { getTeamSettingsMenu } from "../teams/TeamSettings";
+import { PaymentContext } from "../payment-context";
+import FeedbackFormModal from "../feedback-form/FeedbackModal";
+import { inResource, isGitpodIo } from "../utils";
+import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-mode";
+import { FeatureFlagContext } from "../contexts/FeatureFlagContext";
+import OrganizationSelector from "./OrganizationSelector";
+
+interface Entry {
+    title: string;
+    link: string;
+    alternatives?: string[];
+}
+
+export default function Menu() {
+    const user = useCurrentUser();
```

Comment:
One thing I noticed here (and on the team settings sub-menu) - we render a partial nav, then after data completes, we may add additional nav items (like settings and usage). I think it would be a much better experience if we instead loaded the data to determine the full nav menu items, then rendered. This is even more jarring now that they're on the top nav imo. We can utilize react-query to make subsequent renders not need the data calls too. I started something similar for the team settings sub-nav in a PR, and maybe it's a future improvement we can make for the org nav too.

Reply:
Agree. Would love to tackle this ina follow up PR.